go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -343,7 +343,7 @@ func (se *Engine) reload(ctx context.Context) error {
 		// TODO(sougou); find a better way detect changed tables. This method
 		// seems unreliable. The endtoend test flags all tables as changed.
```

Comment:
Let's also update this old comment by Sugu. Shall we clarify here as comment why the logic is as it is? Basically copy+paste the PR comment to explain about table creation and Online DDL etc.